dumpling/export/dump.go

```diff
@@ -379,6 +400,83 @@ func (d *Dumper) dumpDatabases(tctx *tcontext.Context, metaConn *sql.Conn, taskC
 	return nil
 }
 
+// adjustDatabaseCollation adjusts db collation and return new create sql and collation
+func adjustDatabaseCollation(db *sql.Conn, parser1 *parser.Parser, originSQL string, dbName string) (string, string, error) {
+	stmt, err := parser1.ParseOneStmt(originSQL, "", "")
+	if err != nil {
+		return "", "", err
+	}
+	createStmt, ok := stmt.(*ast.CreateDatabaseStmt)
+	if !ok {
+		return originSQL, "", nil
+	}
+	for _, createOption := range createStmt.Options {
+		// already have 'Collation'
+		if createOption.Tp == ast.DatabaseOptionCollate {
+			return originSQL, createOption.Value, nil
+		}
+	}
+	// get db collation
+	collation, err := GetDBCollation(db, dbName)
+	if err != nil {
+		return "", "", err
+	}
+	// add collation
+	createStmt.Options = append(createStmt.Options, &ast.DatabaseOption{Tp: ast.DatabaseOptionCollate, Value: collation})
+	// rewrite sql
+	var b []byte
+	bf := bytes.NewBuffer(b)
+	err = createStmt.Restore(&format.RestoreCtx{
+		Flags: format.DefaultRestoreFlags,
+		In:    bf,
+	})
+	if err != nil {
+		return "", "", err
+	}
+	return bf.String(), collation, nil
+}
+
+// adjustTableCollation adjusts table collation and return new create sql and default db collation
+func adjustTableCollation(db *sql.Conn, parser1 *parser.Parser, originSQL string, dbCollation string, dbName string, tableName string) (string, string, error) {
```

Comment:
`tableName` is never used
```suggestion
func adjustTableCollation(db *sql.Conn, parser1 *parser.Parser, originSQL, dbCollation, dbName string) (string, string, error) {
```